### PR TITLE
Previous source maps support

### PR DESCRIPTION
**Pull request by miripiruni**
Related: 
 * https://github.com/enb/enb-source-map/pull/19
 * https://github.com/enb/enb-bemxjst/pull/284

cc @sbmaxx

### CODE REVIEW

File: lib/bemxjst/match.js [hunk 1]
```diff
@@ -153,6 +153,19 @@ Match.prototype.tryCatch = function(fn, ctx) {
     return fn.call(ctx, ctx, ctx.ctx);
   } catch (e) {
     this.thrownError = e;
+    if (this.modeName) {
+      this.thrownError.ctx = ctx;
+      this.thrownError.name = 'BEMXJST ERROR';
+      var classBuilder = this.entity.bemxjst.classBuilder;
+
+      var cause = e.stack.split('\n')[1];
+      this.thrownError.message = 'Ошибка при шаблонизации режима ' +
```
Comment (sbmaxx) [inline lib/bemxjst/match.js]: Maybe this should be written in english? :)

**Comment by miripiruni** [comment]
@zxqfox @tadatuta review needed

**Comment by sbmaxx** [comment]
Let's merge this ;)

File: lib/compiler.js [hunk 2]
```diff
@@ -94,18 +95,13 @@ Compiler.prototype.generate = function generate(code, options) {
...
-      // Provides third-party libraries from different modular systems
-      'BEMContext.prototype.require = function(lib) {',
-       'return this._libs[lib];',
-      '};',
-    '});'
-  ].join('');
+  code = fnToString(code);
 
   var deps = getDeps(options.requires);
+  var bemxjstOpts = utils.extend({}, options);
```
Comment (tadatuta) [inline lib/compiler.js]: any reason to use custom `extend` instead of `Object.assign()` here?

File: docs/en/3-api.md [hunk 3]
```diff
@@ -572,8 +572,7 @@ There is options in `generate` method to use source maps.
 
 * `to` {String} — output bundle file name
 * `sourceMap.from` {String} — file name
-* `sourceMap.prev` {String|Function|SourceMapConsumer|SourceMapGenerator} —
-* previous source maps in any format
+* `sourceMap.prev` {Object} — previous source map object
```
Comment (tadatuta) [inline docs/en/3-api.md]: isn't it major change?

Comment (miripiruni) [reply docs/en/3-api.md]: Not at all.

File: lib/compiler.js (hunk 2 above)
Comment (miripiruni) [reply lib/compiler.js]: @tadatuta replaced. Thanks!

Comment (miripiruni) [reply lib/compiler.js]: @tadatuta let’s drop all not actual LTS Node.js support in next major version.

Comment (sbmaxx) [reply lib/compiler.js]: >  Ah, we still support node 0.12 =)

🙀

Comment (miripiruni) [reply lib/compiler.js]: @sbmaxx well, because technicaly it’s a major release =)

**Pull request merged**